capa/features/address.py

```diff
@@ -36,6 +36,58 @@ def __hash__(self):
         return int.__hash__(self)
 
 
+class ProcessAddress(Address):
+    """addresses a processes in a dynamic execution trace"""
+
+    def __init__(self, pid: int, ppid: int = 0):
+        assert ppid >= 0
+        assert pid > 0
+        self.ppid = ppid
+        self.pid = pid
+
+    def __repr__(self):
+        return "process(%s%s)" % (
+            f"ppid: {self.ppid}, " if self.ppid > 0 else "",
+            f"pid: {self.pid}",
+        )
+
+    def __hash__(self):
+        return hash((self.ppid, self.pid))
+
+    def __eq__(self, other):
+        assert isinstance(other, ProcessAddress)
+        if self.ppid > 0:
```

Comment:
im afraid this check here might lead to some surprising behavior, if one has ppid set and the other does not. are you sure this cannot be simplified to `return (self.pid, self.ppid) == (other.pid, other.ppid)`?